### ISSUE TITLE
Upgrade Twig

### ISSUE THREAD

**Issue by karolyka** [trampgeek/moodle-qtype_coderunner#81]
Do you plan to upgrade Twig to the latest version?

**Comment by trampgeek** [comment]
I'm reluctant to upgrade to the latest Twig because it needs at least PHP 7.1.3. I think there are quite a lot of Moodle sites that run on older servers which still use PHP 6 or 7.0. There would need to be a very significant benefit before I upgraded. What features of Twig 2 did you want to use?

**Comment by karolyka** [comment]
Thank you for your answer! So, There are many new features in Twig after the included 1.28.2 version. The map function, sort with non-default order, reverse, etc. I know, there are many features released in 1.4, 1.6 versions not the 2.x, but I think the newest version has the most features. I pulled your sourcecode and I modified for the newest Twig. I made some another modification too and a new functiononality for SQL server. If you think it may be useful to you, I will be happy to send or upload it.
